Fix(Topofit): Fixed Docker Image error

gaiborjosue opened the pull request:
This PR will successfully close #3 @hvgazula

gaiborjosue commented:
The action will fail on my end because I don't have docker credentials set up in the repo. However, the image was built successfully. In other words, the only step that failed on my end was pushing the image.

hvgazula commented:
@gaiborjosue I too got it to work with 0.6.0 myself. But I am trying to stay true to the user's setup. I spoke to @williamashbee yesterday and he was pretty sure he did not have any issues with building an image with the Dockerfile as is.

gaiborjosue commented:
@hvgazula I see. I will take a look on that.

WilliamAshbee commented:
I will see if i can get topofit to build again on my end. maybe i have some local changes or maybe something was deprecated. at one point topofit was pretty smooth build.

gaiborjosue commented:
Hello @WilliamAshbee, that would be helpful. I tried installing surfa==0.0.8 locally, and it did not work either.

WilliamAshbee commented:
It looks like surfa==0.0.8 is not in pypi anymore (not sure how that happened), but I removed this line from the environment.yml and did a 'RUN pip install surfa'  in the docker file and the build finishes. @hvgazula  and I discussed this earlier today.

gaiborjosue commented:
Awesome. Thanks!

Edward Gaibor
________________________________
From: William Stewart Ashbee ***@***.***>
Sent: Wednesday, August 2, 2023 4:14:34 PM
To: hvgazula/build_docker ***@***.***>
Cc: Edward Gaibor ***@***.***>; Mention ***@***.***>
Subject: Re: [hvgazula/build_docker] Fix(Topofit): Fixed Docker Image error (PR #4)


It looks like surfa==0.0.8 is not in pypi anymore (not sure how that happened), but I removed this line from the environment.yml and did a 'RUN pip install surfa' in the docker file and the build finishes. @hvgazula<https://github.com/hvgazula> and I discussed this earlier today.

—
Reply to this email directly, view it on GitHub<https://github.com/hvgazula/build_docker/pull/4#issuecomment-1662972996>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASZOEDJDYLQMN5D45DMHR3DXTK7LVANCNFSM6AAAAAA3BL6NDE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

hvgazula commented:
> It looks like surfa==0.0.8 is not in pypi anymore (not sure how that happened), but I removed this line from the environment.yml and did a 'RUN pip install surfa' in the docker file and the build finishes. @hvgazula and I discussed this earlier today.

Thanks @WilliamAshbee. Is this your final go-ahead (aka did you get to run some preliminary tests to ensure that the results are just the same)?

hvgazula commented:
Another option that I just realized is to upload your earliest image (with 0.0.8) and we will use that in the zoo. what do you think @WilliamAshbee?

WilliamAshbee commented:
@hvgazula I never pushed to dockerhub unfortunately. Our local storage doesn't usually last more than a few weeks. Sorry, I would think the most recent version is surfa would work. If you would like I can perform some testing.

hvgazula commented:
> @hvgazula I never pushed to dockerhub unfortunately. Our local storage doesn't usually last more than a few weeks. Sorry, I would think the most recent version is surfa would work. If you would like I can perform some testing.

I thought that's what you wanted to do. But, again only if you have time. If not, we will proceed with it as is (with the new modification) with no guarantees provided for future results. 😄

WilliamAshbee commented:
I would proceed as is, and in parallel i will test it.

WilliamAshbee commented:
I have updated the dockerfile. The versions of all dependencies has remained the same. Had to use anaconda instead of pip for torch-scatter. Had to compile surfa from source, but is same version as tested version. Had @MiaLinLinLu verify that it matches the review paper's results, so it reproduces results on the test set. The github fork has the updated dockerfile, yml, and surfa source (small). https://github.com/neuroneural/topofit_fork

hvgazula commented:
Thank you. I built all the images this morning and they should be in my
docker hub.

On Fri, Aug 4, 2023 at 9:13 AM William Stewart Ashbee <
***@***.***> wrote:

> I have updated the dockerfile. The versions of all dependencies has
> remained the same. Had to use anaconda instead of pip for torch-scatter.
> Had to compile surfa from source, but is same version as tested version.
> Had @MiaLinlinLu <https://github.com/MiaLinlinLu> verify that it matches
> the review paper's results, so it reproduces results on the test set. The
> github fork has the updated dockerfile, yml, and surfa source (small).
> https://github.com/neuroneural/topofit_fork
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hvgazula/build_docker/pull/4#issuecomment-1665590026>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACHXDP7ONGXBHOT2OYEIPFTXTTYQFANCNFSM6AAAAAA3BL6NDE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 

*Thanks & Regards,*
*Harsha Gazula*
